keys/openidprovider.go

```diff
@@ -3,29 +3,25 @@ package keys
 import (
 	"encoding/json"
 	"fmt"
-	"github.com/coreos/dex/pkg/log"
+	"github.com/zalando/planb-tokeninfo/breaker"
+	"github.com/zalando/planb-tokeninfo/options"
 	"io/ioutil"
-	"net/http"
-	"os"
+	"log"
+	"net/url"
 	"reflect"
-	"time"
 )
 
 // http://openid.net/specs/openid-connect-discovery-1_0.html#ProviderConfig
 // https://planb-provider.example.org/.well-known/openid-configuration
+// https://accounts.google.com/.well-known/openid-configuration
 type cachingOpenIdProviderLoader struct {
 	url      string
 	keyCache *Cache
 }
 
-const defaultRefreshInterval = 30 * time.Second
-
-const OPENID_PROVIDER_CONFIGURATION_URL = "OPENID_PROVIDER_CONFIGURATION_URL"
-
-func newCachingOpenIdProviderLoader() KeyLoader {
-	u := os.Getenv(OPENID_PROVIDER_CONFIGURATION_URL)
-	kl := &cachingOpenIdProviderLoader{url: u, keyCache: NewCache()}
-	schedule(defaultRefreshInterval, kl.refreshKeys)
+func NewCachingOpenIdProviderLoader(u *url.URL) KeyLoader {
+	kl := &cachingOpenIdProviderLoader{url: u.String(), keyCache: NewCache()}
```

Comment:
since now this is public, it would be nice to check if u is not nil. just cosmetics, though.